diff --git internal/npm_install/npm_install.bzl internal/npm_install/npm_install.bzl
index adfcef1..dc1e048 100755
--- internal/npm_install/npm_install.bzl
+++ internal/npm_install/npm_install.bzl
@@ -822,16 +822,6 @@ def _yarn_install_impl(repository_ctx):
     if repository_ctx.attr.frozen_lockfile:
         yarn_args.append("--frozen-lockfile")

-    if not repository_ctx.attr.use_global_yarn_cache:
-        yarn_args.extend(["--cache-folder", str(repository_ctx.path("_yarn_cache"))])
-    else:
-        # Multiple yarn rules cannot run simultaneously using a shared cache.
-        # See https://github.com/yarnpkg/yarn/issues/683
-        # The --mutex option ensures only one yarn runs at a time, see
-        # https://yarnpkg.com/en/docs/cli#toc-concurrency-and-mutex
-        # The shared cache is not necessarily hermetic, but we need to cache downloaded
-        # artifacts somewhere, so we rely on yarn to be correct.
-        yarn_args.extend(["--mutex", "network"])
     yarn_args.extend(repository_ctx.attr.args)

     # Run the package manager in the package.json folder
